.github/workflows/catalog.yml

```diff
@@ -1,4 +1,9 @@
-on: [push]
+on:
+  push:
+    paths:
+      - '.github/workflows/catalog.yml'
+      - 'catalog/**'
+      - 'shared/**'
```

Comment:
I think that doesn't work well with required jobs (you can check them (you can see them in repo settings -> branches -> master)
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions
> If a workflow is skipped due to branch filtering, [path filtering](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore), or a [commit message](https://docs.github.com/en/actions/managing-workflow-runs/skipping-workflow-runs), then checks associated with that workflow will remain in a "Pending" state. A pull request that requires those checks to be successful will be blocked from merging.

you can check registry workflow in enterprise for idea on how to implement filtering

also I suggest testing this CI stuff in a fork